new_lessons/sets.md

```diff
@@ -0,0 +1,337 @@
+
+```python
+
+set_a = {1, 2, 3}
+set_b = {1, 2, 3, 4, 5}
+
+# Only one syntax methods to test if a set is a subset of another set
+# < operator
+set_a < set_b # evaluates to True
+
+# A set is not a proper subset of another set if they are equal
+set_c = {1, 2, 3}
+
+set_a < set_c # evaluates to False
+```
+
+#### Disjoint
+
+Two sets are disjoint if they have no overlapping elements.
+
+```python
+
+set_a = {1, 2, 3}
+set_b = {4, 5, 6}
+set_c = {3, 4, 5}
+
+# Only one syntax method to test if a set is disjoint with another set
+# disjoint() function
+set_a.disjoint(set_b) # evaluates to True
+set_a.disjoint(set_c) # evaluates to False
+```
+
+## Sets and Problem Solving
+
+Now that we know set basics, let's take a look at how we can use sets to solve problems!  
+
+Lets say we are working on an algorithm to compare two lists and find the elements that are common to both.  If we decided to approach this with a brute force solution, we might start by looping over all of the elements in the first list.  Then, for each element in the first list we would search inside the second list and see if we can find that element from the first list in the second list.  If we do find the element in the second list, we would add it to a result list.  Here's the code for this type of approach:
+
+```python
+
+list_a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+list_b = [6, 7, 8, 9, 10, 11, 12, 13, 14]
+
+result_list = []
+for element_a in list_a:
+    if element_a in list_b:
+        result_list.append(element_a)
+
+return result_list
+```
+
+This solution will get the job done, but it's not very efficient or elegant!  We can use sets to simplify this code by converting our lists into sets and then using set functionality to solve the problem.
+
+### Creating Sets From Other Data Structures
+
+First we need to turn our lists into sets.  The syntax for this is:
+
+```python
+
+list_a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+list_b = [6, 7, 8, 9, 10, 11, 12, 13, 14]
+
+set_a = set(list_a) # converts the list into a set!
+set_b = set(list_a) 
+
+```
+
+Next, we can use set intersection to generate a set of all of the elements that are contained in both sets:
+
+```python
+
+result_set = set_a & set b
+
+```
+
+If we want the results in a list we can convert a set into a list in a similar way to how we converted the list into a set:
+
+```python
+
+result_list = list(result_set)
+
+```
+
+Here's the final version of the code:
+
+```python
+
+list_a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+list_b = [6, 7, 8, 9, 10, 11, 12, 13, 14]
+
+set_a = set(list_a) # converts the list into a set!
+set_b = set(list_a) 
+
+result_set = set_a & set b
+
+return list(result_set)
+
+```
+
+### Iterating Through Sets
+
+Let's say we have a string of letters and we want to find all of the unique characters in the string.  Here's a brute force solution that we could use:
+
+```python
+
+s = "she sells sea shells by the sea shore"
```

Comment:
Since there's a lot of "s" words (set... string... she sells sea shells...) could we find a different variable name besides `s`?